### ISSUE TITLE
Add bulk update/delete operations to API

### ISSUE THREAD

**Issue by rafalp** [rafalp/Misago#884]
When user selects two or more posts on thread view and selects "delete posts" action, following error is occasionally raised:

```
OperationalError: deadlock detected
DETAIL:  Process 25972 waits for ShareLock on transaction 6914969; blocked by process 25973.
Process 25973 waits for ShareLock on transaction 6914968; blocked by process 25972.
HINT:  See server log for query details.
CONTEXT:  while updating tuple (9,44) in relation "misago_threads_thread"
```

This is likely to occur for posts move as well.

Upon some debugging I've came to conclusion that best way to go about this issue would be implementation of massmoderation endpoints for threads and posts.

**Bulk ops**

* [x] Bulk delete posts
* [x] Bulk delete threads
* [x] Bulk patch posts
* [x] Bulk patch threads
* [x] 404 on bulk request in endpoints that don't support those

**Locking**

Instead of locking other tables, we'll use `request.user.lock()` in critical endpoints (like/vote) to protect against flood by individual user and remove locking from other endpoints. We'll still wrap some ops in transactions to minimize risk of data corruption (eg. poll creation poll/post vote, thread move, delete).

**Comment by rafalp** [comment]
This issue is not marked as critical due to PostgreSQL's deadlock handling causing no data loss.

**Comment by rafalp** [comment]
So here's realization:

You may send PATCH/PUT/UPDATE/DELETE to Django Rest Framework without passing PK in URL, thus letting you look up the `request.data` and handle mass action.

...thus all our endpoints will happily 500 on maliciously constructed requests to update/partial_update/delete endpoints powered by Misago.

**Comment by rafalp** [comment]
Backend on this change is mostly done now, I just need to redo transactions/locks and update frontend JS to use those bulk backends

**Comment by rafalp** [comment]
todo: close this issue and split locking into separate issue.
